js/ui/appActivation.js

```diff
@@ -155,7 +157,7 @@ const AppActivationContext = new Lang.Class({
             this._abort = true;
     },
 
-    showSplash: function() {
+    showSplashIfNeeded: function() {
```

Comment:
I would not rename this method, since even before your changes it did not show the splash when not needed.

Reply:
Yes, I know. And I found that particularly confusing so I thought it would be nice to "fix" that.